app/reducers/roles.js

```diff
@@ -0,0 +1,28 @@
+import { ROLES } from '../actions/actionsTypes';
+
+const initialState = {
+	roles: {}
+};
+
+export default function permissions(state = initialState, action) {
+	switch (action.type) {
+		case ROLES.SET:
+			return action.roles;
+		case ROLES.UPDATE:
+			return {
+				...state,
+				[action.payload.id]: action.payload.desc || action.payload.id
+			};
+		case ROLES.ADD:
+			return {
+				...state,
+				...{ [action.payload.id]: action.payload.desc }
+			};
+		case ROLES.REMOVE:
+			delete state[action.payload.id];
```

Comment:
You should never mutate the state.
Try something like this:
```
return state.filter(element => element !== action.payload);
```

https://stackoverflow.com/questions/34582678/is-this-the-correct-way-to-delete-an-item-using-redux